ui/src/pages/version/components/forms/components/ProtocolDropdown.js

```diff
@@ -0,0 +1,65 @@
+import React, { Fragment } from "react";
+import { EuiSuperSelect, EuiText } from "@elastic/eui";
+
+/**
+ * Dropdown menu for selecting protocol
+ *
+ * @param {*} endpointStatus current endpoint status
+ * @param {*} protocol current selected protocol
+ * @param {*} onProtocolChange callback to be made when protocol selection changed
+ *
+ */
+export const ProtocolDropdown = ({
+  endpointStatus,
+  protocol,
+  onProtocolChange,
+}) => {
+  // Users should not be able to change protocol when the model endpoint status is serving
+  const isDisabled = (endpointStatus) => {
```

Comment:
user can change protocol?

Reply:
Yes, as long as it's not in `serving` state. Is there any concern?

Reply:
i think this contradict with this https://github.com/gojek/merlin/pull/273#discussion_r943234493

Reply:
That comment is more of the restriction in model endpoint. In our design model endpoint can have several version endpoint (although it's currently not possible), and that code is to prevent model endpoint from having version endpoint with different protocol.

Reply:
got it